src/types.ts

```diff
@@ -0,0 +1,161 @@
+import Key from './key'
+
+export type AnyIterable<T> = Iterable<T> | AsyncIterable<T>
```

Comment:
I kept calling this `AwaitIterable<T>` in other places which I think is a better name (as it implies that you need to await as you iterate)